### ISSUE TITLE
Node compilation

### ISSUE THREAD

**Issue by noahlange** [noahlange/superjucks#3]
Obviously each node (~35) will need a `Node#compile()` function before this gets off the ground.

Many of them (binary ops + unary ops) are pretty straightforward.
Many of them (mostly tags) are not.
Asterisk'd nodes also need a parser method.

## Tags
- [ ] Block
- [x] Case
- [ ] Export
- [ ] Extends*
- [x] Filter (block)
- [x] For
- [x] If
- [ ] Import
- [ ] Include
- [x] Macro
- [x] Raw*
- [x] Switch
- [x] Unless

## Utils
- [x] Aggregate
- [x] Array
- [x] Assign
- [x] Capture
- [x] CompareOperand
- [x] Dict
- [x] FunctionCall
- [x] Literal
- [x] LookupVal
- [x] Output
- [x] Root
- [x] Symbol

## Operators
- [x] Add
- [x] And
- [x] Await
- [x] Compare
- [x] Concat
- [x] Div
- [x] Elvis
- [x] Filter (operator)
- [x] FloorDiv
- [x] In
- [x] Is
- [x] List
- [x] Mod
- [x] Mul
- [x] Not
- [x] NullCoalescing
- [x] Or
- [x] Pair
- [x] Pow
- [x] Range
- [x] Set
- [x] Sub
- [x] Ternary

**Comment by noahlange** [comment]
Partially addressed in [feat-node-compilation](https://github.com/noahlange/superjucks/tree/feat-node-compilation).

**Comment by ghost** [comment]
that branch doesn't exist. I think you've done an amazing job with this variant so far though.

**Comment by noahlange** [comment]
Yeah, I think it got merged in at some point. Thank you so much, though. Been working on it on-and-off.

**Comment by ghost** [comment]
so what is actually left before it's ready to try out?

**Comment by noahlange** [comment]
Well, I need to split the package into runtime and compile-time modules.

And at some point in the last six months, likely inspired by Twig, I started refactoring the root node into an ES6 class to make inheritance a little more idiomatic.

```jinja
{% extends 'tpl.sjk' %}

{% block foo %}
  {{ super() }}
  Oh hi, Mark!
{% endblock %}
```

```js
export default _ =>
  class Template__3bqekeki1 extends _.template('tpl.sjk') {
    static get id() {
      return '_3bqekeki1';
    }
    async foo() {
      const buffer = new _.Buffer();
      buffer.write(await super.foo());
      buffer.write('Oh hi, Mark!');
      return _.exports(buffer.out());
    }
  }
```

And that might be a pretty significant overhaul. 😕I'm constantly juggling a few other projects, so it might be a bit before I cycle back to this one. Sorry. 

If you've got a specific use-case I could *probably* make it happen a little sooner, but otherwise it might be a couple months before I refocus my attention on this.
